catroid/src/androidTestCreateatschool/java/org/catrobat/catroid/createatschool/uiespresso/ui/activity/RestrictedLoginTest.java

```diff
@@ -0,0 +1,232 @@
+
+		UiTestUtils.setFakeToken(baseActivityTestRule.getActivity(), false);
+	}
+
+	@Test
+	public void loginDialogsUiBehaviorTest() {
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(matches(isDisplayed()));
+		onView(withText(R.string.cancel))
+				.check(doesNotExist());
+		pressBack();
+
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(matches(isDisplayed()));
+
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		onView(allOf(withText(R.string.login), withResourceName("alertTitle")))
+				.check(matches(isDisplayed()));
+		onView(withText(R.string.password_forgotten))
+				.check(doesNotExist());
+		onView(withText(R.string.cancel))
+				.check(doesNotExist());
+
+		pressBack();
+
+		onView(allOf(withText(R.string.login), withResourceName("alertTitle")))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void restrictedLoginDialogLinkTest() {
+		String link = Locale.getDefault().getLanguage().equals(Locale.GERMAN.getLanguage())
+				? Constants.RESTRICTED_LOGIN_LINK_GERMAN
+				: baseActivityTestRule.getActivity().getString(R.string.schoolreg_link);
+
+		Intents.init();
+		Matcher<Intent> expectedIntent = allOf(hasAction(Intent.ACTION_VIEW), hasData(link));
+		intending(expectedIntent).respondWith(new Instrumentation.ActivityResult(0, null));
+		Espresso.onView(ViewMatchers.withId(R.id.dialog_restricted_login_text))
+				.perform(ViewActions.openLinkWithText(link));
+		intended(expectedIntent);
+		Intents.release();
+	}
+
+	@Test
+	public void restrictedLoginDialogNotShownAfterConfirmationTest() {
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		relaunchActivity();
+
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(doesNotExist());
+	}
+
+	@Test
+	public void restrictedLoginDialogReappearsAfterLogoutTest() {
+		UiTestUtils.setFakeToken(baseActivityTestRule.getActivity(), true);
+		relaunchActivity();
+
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(doesNotExist());
+		onView(allOf(withText(R.string.login), withResourceName("alertTitle")))
+				.check(doesNotExist());
+
+		openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getTargetContext());
+		onView(withText(R.string.main_menu_logout))
+				.perform(click());
+
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void loginWithOtherThanCreateAtSchoolUserTest() {
+		ServerCalls.useTestUrl = true;
+		String testUserName = "testUser" + System.currentTimeMillis();
+		String testPassword = "password";
+		String testUserMail = testUserName + "@catrob.at";
+		UiTestUtils.createValidUserWithCredentials(baseActivityTestRule.getActivity(), testUserName, testPassword, testUserMail);
+
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		onView(withId(R.id.dialog_login_username))
+				.perform(clearText(), typeText(testUserName));
```

Comment:
add closeSoftKeyboard() as additional action after typeText.
i found that tests become increasingly flake and some really hard to find errors happen because sometimes the softkeyboard stays open and covers the next uielement you want to access.

Reply:
good hint